controllers/cinder_controller.go

```diff
@@ -430,6 +553,61 @@ func (r *CinderReconciler) reconcileNormal(ctx context.Context, instance *cinder
 			instance.Status.CinderVolumesReadyCounts = map[string]int32{}
 		}
 		instance.Status.CinderVolumesReadyCounts[name] = cinderVolume.Status.ReadyCount
+
+		// Check if the CinderVolume has a "True" condition with an error reason, as we will
+		// prioritize recording/displaying the first error condition encountered as the
+		// CinderVolumeReadyCondition for the Cinder CR (this will take precedence over any
+		// sort of overall condition for the CinderVolumes)
+		c = cinderVolume.Status.Conditions.Mirror(cinderv1beta1.CinderVolumeReadyCondition)
+		if c != nil {
+			if c.Reason == condition.ErrorReason && volumeErrorCondition == nil {
+				// If there is an error and we didn't already find another error on another
+				// CinderVolume, record this error for use down below
+				volumeErrorCondition = c.DeepCopy()
+				// No need to consider the final "volumesReady" if-statement now,
+				// as this error takes precedence
+				continue
+			}
+		}
+
+		if cinderVolume.Status.Conditions.IsTrue(condition.DeploymentReadyCondition) {
+			volumesReady++
+		}
+	}
+
+	if volumeErrorCondition != nil {
+		// If there was an explicit error condition, set that as the CinderVolumeReadyCondition
+		instance.Status.Conditions.Set(volumeErrorCondition)
+	} else if volumesReady == len(instance.Spec.CinderVolumes) {
+		// The CinderVolumes are ready.
+		// Using "condition.DeploymentReadyMessage" here because that is what gets mirrored
+		// as the message for the other Cinder children when they are successfully-deployed
+		instance.Status.Conditions.MarkTrue(cinderv1beta1.CinderVolumeReadyCondition, condition.DeploymentReadyMessage)
+	} else {
+		// CinderVolumes are still in the process of deploying.
+		// As they all could have varying non-error conditions, we just use an overall
+		// "CinderVolume deployment in progress" message here
+		instance.Status.Conditions.Set(condition.FalseCondition(
```

Comment:
I'm not sure I understand.  If we have multiple `CinderVolumes`, how do we pick which one to mirror for the non-error case?

Reply:
how about we simplify all of this with something like the following. It depends on https://github.com/openstack-k8s-operators/lib-common/pull/48 which allows to compare two conditions to get the one which has precedence. 

```
...
	var volumeCondition *condition.Condition
	for name, volume := range instance.Spec.CinderVolumes {
		cinderVolume, op, err := r.volumeDeploymentCreateOrUpdate(instance, name, volume)
		if err != nil {
			instance.Status.Conditions.Set(condition.FalseCondition(
				cinderv1beta1.CinderVolumeReadyCondition,
				condition.ErrorReason,
				condition.SeverityWarning,
				cinderv1beta1.CinderVolumeReadyErrorMessage,
				err.Error()))
			return ctrl.Result{}, err
		}
		if op != controllerutil.OperationResultNone {
			r.Log.Info(fmt.Sprintf("Deployment %s successfully reconciled - operation: %s", instance.Name, string(op)))
		}

		// Mirror CinderVolume status' ReadyCount to this parent CR
		// TODO: Somehow this status map can be nil here despite being initialized
		//       in the Reconcile function above
		if instance.Status.CinderVolumesReadyCounts == nil {
			instance.Status.CinderVolumesReadyCounts = map[string]int32{}
		}
		instance.Status.CinderVolumesReadyCounts[name] = cinderVolume.Status.ReadyCount

		// If this cinderVolume is not IsReady, mirror the condition to get the latest step it is in.
		// Could also check the overall ReadyCondition of the cinderVolume.
		if !cinderVolume.IsReady() {
			c = cinderVolume.Status.Conditions.Mirror(cinderv1beta1.CinderVolumeReadyCondition)
			// Get the condition with higher priority for volumeCondition.
			volumeCondition = condition.GetHigherPrioCondition(c, volumeCondition).DeepCopy()
		}
	}

	if volumeCondition != nil {
		// If there was a Status=False condition, set that as the CinderVolumeReadyCondition
		instance.Status.Conditions.Set(volumeCondition)
	} else {
		// The CinderVolumes are ready.
		// Using "condition.DeploymentReadyMessage" here because that is what gets mirrored
		// as the message for the other Cinder children when they are successfully-deployed
		instance.Status.Conditions.MarkTrue(cinderv1beta1.CinderVolumeReadyCondition, condition.DeploymentReadyMessage)
	}

	r.Log.Info("Reconciled Service successfully")
	return ctrl.Result{}, nil
```

Reply:
That's a good idea

Reply:
Changed to use this approach